### PR TITLE
Beginnings of oauth.

### PR DESCRIPTION
**Pull request by alexcouper**
Missing the end of the dance...

### CODE REVIEW

File: ployst/github/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from .conf import settings
+
+from ployst.core.client import Client
+
+client = Client('http://localhost:8000/', settings.GITHUB_CORE_API_TOKEN)
```
Comment (txels) [inline ployst/github/__init__.py:5]: shouldn't this URL be a setting as well?

File: ployst/github/test/test_oauth.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    @override_settings(GITHUB_OAUTH_STATE='jill')
+    def test_oauth_url_redirects_correctly(self):
+        "Calls to start oauth should redirect to github"
+        response = self.client.get(reverse('github:oauth-start'))
+        self.assertEquals(response.status_code, 302)
+        self.assertEquals(
+            response['location'],
+            ('https://github.com/login/oauth/authorize?'
+             'client_id=fred&'
+             'scope=repo,write:repo_hook&'
```
Comment (txels) [inline ployst/github/test/test_oauth.py:25]: we may want `write:public_key` eventually? Won't we need that for repo cloning?

File: ployst/github/test/test_oauth.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        self.assertEquals(
+            set_access_token.call_args[0],
+            ('github', access_token)
+        )
+
+    def test_github_returns_error(self):
+        """
+        Test what happens when github comes back with some non-200 status.
+
+        We expect to LOG this sitation for now as the docs don't give us any
```
Comment (txels) [inline ployst/github/test/test_oauth.py:93]: situation

File: requirements/test.txt [hunk 4]
```diff
@@ -4,3 +4,4 @@ factory-boy==2.3.0
 mock
 nose==1.3.0
 supermutes
+httpretty
```
Comment (txels) [inline requirements/test.txt:7]: keep sorted? 
also eventually I like to add a comment to every non-obvious dependency, which we haven't done here.

File: ployst/core/client.py [hunk 5]
```diff
@@ -100,3 +100,7 @@ def create_or_update_branch_information(self, branch_info):
                      branch_info)
         else:
             self.post('repos/branch/', branch_info)
+
+    def set_access_token(oauth_provider, access_token):
```
Comment (txels) [inline ployst/core/client.py]: missing `self`

File: ployst/github/views/oauth.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    )
+
+
+@require_http_methods(['GET'])
+def receive(request):
+    """End point to receive the redirect back from github"""
+    if request.GET['state'] != settings.GITHUB_OAUTH_STATE:
+        return HttpResponseBadRequest()
+    exchange_for_access_token(request.GET['code'])
+    return HttpResponse('OK')
```
Comment (txels) [inline ployst/github/views/oauth.py]: this is only missing a redirect to a decent location and we're almost done, right? That and properly saving the token for the user...

Comment (alexcouper) [inline ployst/github/views/oauth.py]: Correct

**Comment by alexcouper** [comment]
Cheers for the review, I'll hopefully be able to make the appropriate changes soon...

**Comment by txels** [comment]
I'll merge this so that I can use your flake8 fixes and start to build the provider setup pages as well.

**Pull request merged**